### DISCUSSION TITLE
How does FINN handle quantization and scaling?

### DISCUSSION THREAD

**Discussion by Masoud071** [Xilinx/finn#1007] [Q&A] [answered]
Hello everyone ! I am new to finn and I was reading the docs ( finn.readthedocs.io, finn hls lib) to find out how does it work with quantized weights and inputs exported from brevitas, but I did not see any clear explanation about that. 
I mean what exactly ingested to finn from brevitas? I saw that in hls lib the data type is arbitrary unsigned int and and this question came to me that how does it handles scaling and zero point and do the calculation? 
Thanks in advance for your valuable answers!

**Comment by auphelia** [comment, marked as answer]
Hi @Masoud071,
Please have a look at the cybersecurity tutorial: https://github.com/Xilinx/finn/tree/main/notebooks/end2end_example/cybersecurity
In these series of notebooks, an MLP is trained using Brevitas, then the trained network gets exported and the FINN compiler is used to generate a streaming dataflow accelerator.
